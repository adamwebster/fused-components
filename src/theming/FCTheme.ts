import React from 'react';

export interface Theme {
    theme?: string,
    children?: any,
}

export const FCTheme = React.createContext<Theme | null>(null);

export const FCThemeProvider = FCTheme.Provider;

export const FCThemeConsumer = FCTheme.Consumer;