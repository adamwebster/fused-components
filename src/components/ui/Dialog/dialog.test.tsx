import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest-styled-components';
import { Dialog } from './index';
import { FCThemeProvider } from '../../../theming/FCTheme';

afterEach(cleanup);

describe('Dialog Tests', () => {
  test('Renders the Dialog component', () => {
    const { getByText } = render(
      <Dialog id="1" visible={true}>
        I am a dialog
      </Dialog>,
    );
    expect(getByText('I am a dialog')).toBeInTheDocument();
  });

  test('Clicking the overlay closes the dialog', () => {
    let visible = true;
    const setVisible = jest.fn(() => {
      visible = false;
    });
    const { getByTestId, queryByText, rerender } = render(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    const overlay = getByTestId('dialog-overlay');
    fireEvent.click(overlay);
    expect(visible).toBe(false);

    rerender(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    expect(queryByText('I am a dialog')).toBeFalsy();
  });

  test('Clicking the close button closes the dialog', () => {
    let visible = true;
    const setVisible = jest.fn(() => {
      visible = false;
    });
    const { getByTitle, queryByText, rerender } = render(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    const closeButton = getByTitle('Close dialog');
    fireEvent.click(closeButton);
    expect(visible).toBe(false);

    rerender(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    expect(queryByText('I am a dialog')).toBeFalsy();
  });

  test('Clicking the cancel button closes the dialog', () => {
    let visible = true;
    const setVisible = jest.fn(() => {
      visible = false;
    });
    const { getByText, queryByText, rerender } = render(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    const cancelButton = getByText('Cancel');
    fireEvent.click(cancelButton);
    expect(visible).toBe(false);

    rerender(
      <Dialog id="1" visible={visible} onCloseClick={() => setVisible()}>
        Dialog
      </Dialog>,
    );

    expect(queryByText('I am a dialog')).toBeFalsy();
  });

  test('Setting the box shadow property to false hides the shadow', () => {
    const { getByRole } = render(
      <Dialog id="1" visible boxShadow={false}>
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toHaveStyleRule('box-shadow', undefined);
  });

  test('Able to change the confirm text', () => {
    const { getByText } = render(
      <Dialog id="1" visible confirmText="Click me">
        Dialog
      </Dialog>,
    );
    expect(getByText('Click me')).toBeInTheDocument();
  });

  test('Position is set to relative when fixed is set to false', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fixed={false}>
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toHaveStyleRule('position', 'relative');
  });

  test('Position is set to relative when fixed is set to false', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fixed={false}>
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toHaveStyleRule('position', 'relative');
  });

  test('Has the correct styles fcStyle is set to danger', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fcStyle="danger">
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to warning', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fcStyle="warning">
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to info', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fcStyle="info">
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to success', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true} fcStyle="success">
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles for the default option', () => {
    const { getByRole } = render(
      <Dialog id="1" visible={true}>
        Dialog
      </Dialog>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to danger | dark mode', () => {
    const { getByRole } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Dialog id="1" visible={true} fcStyle="danger">
          Dialog
        </Dialog>
      </FCThemeProvider>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to warning | dark mode', () => {
    const { getByRole } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Dialog id="1" visible={true} fcStyle="warning">
          Dialog
        </Dialog>
      </FCThemeProvider>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to info | dark mode', () => {
    const { getByRole } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Dialog id="1" visible={true} fcStyle="info">
          Dialog
        </Dialog>
      </FCThemeProvider>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles fcStyle is set to success | dark mode', () => {
    const { getByRole } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Dialog id="1" visible={true} fcStyle="success">
          Dialog
        </Dialog>
      </FCThemeProvider>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });

  test('Has the correct styles for the default option | dark mode', () => {
    const { getByRole } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Dialog id="1" visible={true}>
          Dialog
        </Dialog>
      </FCThemeProvider>,
    );
    expect(getByRole('dialog')).toMatchSnapshot();
  });
});
