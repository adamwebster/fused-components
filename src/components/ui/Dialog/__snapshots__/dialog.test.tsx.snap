// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Tests Has the correct styles fcStyle is set to danger | dark mode 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #243b53;
  border: solid 1px #e4585a;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #e4585a;
  box-sizing: border-box;
  position: relative;
  color: #DDDDDD;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c5 {
  border-top: solid 1px #e4585a;
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #627d98;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #4e6479;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to danger 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #fff;
  border: solid 1px #dd2c2e;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #dd2c2e;
  box-sizing: border-box;
  position: relative;
  color: inherit;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c5 {
  border-top: solid 1px #dd2c2e;
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #999999;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #666666;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to info | dark mode 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #243b53;
  border: solid 1px #2f83cf;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #2f83cf;
  box-sizing: border-box;
  position: relative;
  color: #DDDDDD;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c5 {
  border-top: solid 1px #2f83cf;
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #627d98;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #4e6479;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to info 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #fff;
  border: solid 1px #2f83cf;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #2f83cf;
  box-sizing: border-box;
  position: relative;
  color: inherit;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c5 {
  border-top: solid 1px #2f83cf;
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #999999;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #666666;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to success | dark mode 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #243b53;
  border: solid 1px #00854D;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #00854D;
  box-sizing: border-box;
  position: relative;
  color: #DDDDDD;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c5 {
  border-top: solid 1px #00854D;
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #627d98;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #4e6479;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to success 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #fff;
  border: solid 1px #00854D;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #00854D;
  box-sizing: border-box;
  position: relative;
  color: inherit;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c5 {
  border-top: solid 1px #00854D;
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #999999;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #666666;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to warning | dark mode 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #243b53;
  border: solid 1px #eba300;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #eba300;
  box-sizing: border-box;
  position: relative;
  color: #DDDDDD;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c5 {
  border-top: solid 1px #eba300;
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #627d98;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #4e6479;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles fcStyle is set to warning 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #fff;
  border: solid 1px #eba300;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #eba300;
  box-sizing: border-box;
  position: relative;
  color: inherit;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c5 {
  border-top: solid 1px #eba300;
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #999999;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #666666;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles for the default option | dark mode 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #243b53;
  border: solid 1px #627d98;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #627d98;
  box-sizing: border-box;
  position: relative;
  color: #DDDDDD;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c5 {
  border-top: solid 1px #627d98;
  padding: 10px;
  box-sizing: border-box;
  color: #DDDDDD;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #627d98;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #4e6479;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;

exports[`Dialog Tests Has the correct styles for the default option 1`] = `
.c0 {
  border-radius: 5px;
  background-color: #fff;
  border: solid 1px #bcbcbc;
  z-index: 98;
  width: 500px;
  position: fixed;
  margin: 0 auto;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  z-index: 99;
  -webkit-animation: popin 0.2s;
  animation: popin 0.2s;
  box-shadow: 0 0 5px rgba(0,0,0,0.25);
}

.c1 {
  padding: 10px;
  border-bottom: solid 1px #bcbcbc;
  box-sizing: border-box;
  position: relative;
  color: inherit;
}

.c4 {
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c5 {
  border-top: solid 1px #bcbcbc;
  padding: 10px;
  box-sizing: border-box;
  color: inherit;
}

.c2 {
  right: 10px;
  top: 7px;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  font-weight: bold;
  color: #999999;
  position: absolute;
  cursor: pointer;
}

.c2:hover {
  color: #666666;
}

.c2 svg {
  width: 16px;
}

.c3 {
  width: 1em;
  height: 1em;
}

<div
  class="c0"
  id="1"
  role="dialog"
>
  <div
    class="c1"
  >
    Title
    <button
      aria-label="Close"
      class="c2"
      title="Close dialog"
    >
      <svg
        class="c3  times"
        clip-rule="evenodd"
        fill-rule="evenodd"
        role="img"
        stroke-linejoin="round"
        stroke-miterlimit="2"
        title="Close"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 18.343L42.343 0 48 5.657 29.657 24 48 42.343 42.343 48 24 29.657 5.657 48 0 42.343 18.343 24 0 5.657 5.657 0 24 18.343z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <div
    class="c4"
  >
    Dialog
  </div>
  <div
    class="c5"
  >
    Footer
  </div>
</div>
`;
